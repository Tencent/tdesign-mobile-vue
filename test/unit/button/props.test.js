/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { mount } from '@vue/test-utils';
import Button from '../../../src/button';
import { h } from 'vue';

describe('Button', () => {
  describe(':props', () => {
    // 是否为块级元素: false
    it('block', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","block":true},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
    it('content', () => {
      const fn = jest.fn();
      const wrapper = mount(Button, {
        propsData: {
          content: () => h('div', 'text')
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });

    it('content', () => {
      const fn = jest.fn();
      const wrapper = mount(Button, {
        propsData: {
          content: 'text'
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
    it('disabled', () => {
      const fn = jest.fn();
      const wrapper = mount(Button, {
        propsData: {
          disabled: true,
          onClick: fn,
        },
      });
      wrapper.trigger('click');
      expect(fn).not.toHaveBeenCalled();
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 是否禁用按钮: false
    it('disabled', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"按钮1","disabled":true},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 是否为幽灵按钮（镂空按钮）: false
    it('ghost', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","ghost":true},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
    it('icon', () => {
      const fn = jest.fn();
      const wrapper = mount(Button, {
        propsData: {
          icon: () => h('div', 'text')
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });

    it('icon', () => {
      const fn = jest.fn();
      const wrapper = mount(Button, {
        propsData: {
          icon: 'text'
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 是否显示为加载状态: false
    it('loading', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","loading":true},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 按钮形状，有二种：方形、圆角方形 : square
    it('shape', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","shape":"square"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 按钮形状，有二种：方形、圆角方形 : round
    it('shape', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","shape":"round"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 组件尺寸: small
    it('size', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","size":"small"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 组件尺寸: medium
    it('size', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","size":"medium"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 组件尺寸: large
    it('size', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","size":"large"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 组件风格，依次为品牌色、危险色: default
    it('theme', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","theme":"default"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 组件风格，依次为品牌色、危险色: primary
    it('theme', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","theme":"primary"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 组件风格，依次为品牌色、危险色: danger
    it('theme', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","theme":"danger"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 按钮形式，基础、线框、文字: base
    it('variant', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","variant":"base"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 按钮形式，基础、线框、文字: outline
    it('variant', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","variant":"outline"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 按钮形式，基础、线框、文字: text
    it('variant', () => {
      const wrapper = mount(Button, {
        propsData: {"content":"我是按钮","variant":"text"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
  });
});