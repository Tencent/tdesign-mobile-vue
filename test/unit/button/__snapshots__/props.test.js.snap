// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button :props block 1`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--default t-button--square t-size-full-width"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props content 1`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--default t-button--square"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    <div>
      text
    </div>
  </span>
</button>
`;

exports[`Button :props content 2`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--default t-button--square"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    text
  </span>
</button>
`;

exports[`Button :props disabled 1`] = `
<button
  aria-disabled="true"
  class="t-button t-size-m t-button--base t-button--default t-button--square t-is-disabled"
  disabled=""
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    <!---->
  </span>
</button>
`;

exports[`Button :props disabled 2`] = `
<button
  aria-disabled="true"
  class="t-button t-size-m t-button--base t-button--default t-button--square t-is-disabled"
  disabled=""
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    按钮1
  </span>
</button>
`;

exports[`Button :props ghost 1`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--default t-button--square t-button--ghost"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props icon 1`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--default t-button--square"
  role="button"
>
  <div>
    text
  </div>
  <span
    class="t-button__text"
  >
    <!---->
  </span>
</button>
`;

exports[`Button :props icon 2`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--default t-button--square"
  role="button"
>
  text
  <span
    class="t-button__text"
  >
    <!---->
  </span>
</button>
`;

exports[`Button :props loading 1`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--default t-button--square t-is-loading"
  role="button"
>
  <svg
    class="t-icon t-icon-loading"
    fill="none"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <path
      d="M8 1.5a6.5 6.5 0 000 13v-1.63A4.87 4.87 0 1112.88 8h1.62A6.5 6.5 0 008 1.5z"
      fill="currentColor"
      fill-opacity="0.9"
    />
  </svg>
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props shape 1`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--default t-button--square"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props shape 2`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--default t-button--round"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props size 1`] = `
<button
  aria-disabled="false"
  class="t-button t-size-s t-button--base t-button--default t-button--square"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props size 2`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--default t-button--square"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props size 3`] = `
<button
  aria-disabled="false"
  class="t-button t-size-l t-button--base t-button--default t-button--square"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props theme 1`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--default t-button--square"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props theme 2`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--primary t-button--square"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props theme 3`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--danger t-button--square"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props variant 1`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--base t-button--default t-button--square"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props variant 2`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--outline t-button--default t-button--square"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;

exports[`Button :props variant 3`] = `
<button
  aria-disabled="false"
  class="t-button t-size-m t-button--text t-button--default t-button--square"
  role="button"
>
  <!---->
  <span
    class="t-button__text"
  >
    我是按钮
  </span>
</button>
`;
